aws-janitor/resources/s3_bucket.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Bucket struct{}
+
+// IsManagedS3Bucket checks whether the bucket should be managed by janitor.
+func IsManagedS3Bucket(opts Options, region string, bucketName string) (bool, error) {
+	svc := s3.New(opts.Session, &aws.Config{Region: aws.String(region)})
+	respTags, err := svc.GetBucketTagging(&s3.GetBucketTaggingInput{Bucket: aws.String(bucketName)})
+	if err != nil {
+		return false, err
+	}
+	tags := fromS3Tags(respTags.TagSet)
+	return opts.ManagedPerTags(tags), nil
+}
+
+func (S3Bucket) MarkAndSweep(opts Options, set *Set) error {
+	logger := logrus.WithField("options", opts)
+	if !opts.EnableS3BucketClean {
+		logger.Info("Disable s3 bucket clean")
```

Comment:
Nit: maybe something like "s3 bucket cleanup is not enabled".  Or is the idea that we shouldn't get here because the set should be empty when `opts.EnableS3BucketClean` is not true?

Reply:
We will still get here when the set is empty, 
https://github.com/kubernetes-sigs/boskos/blob/62e7de09c721092f127e01268234897a0b2f3d6b/aws-janitor/resources/clean.go#L58